### PR TITLE
Add mark and unmark command

### LINKED ISSUE #96: Implement mark and unmark for attendance taking

### PR DESCRIPTION
**Pull request by SwaminathanViswa**
When marking: `mark n/[NAME] w/[WEEK NUMBER]`
Screenshot 2024-10-22 220100

When unmarking: `unmark n/[NAME] w/[WEEK NUMBER]`
image

Relevant UI changes have also been made.
Closes #96

### CODE REVIEW

**Review by DesSnowy — COMMENTED** [review]
Looks good, just some extra checks that I think is better if added.

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            + "Parameters: " + PREFIX_NAME + "NAME" + PREFIX_WEEK + "WEEK NUMBER"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_WEEK + "1";
+
+    public final int week;
+    private final Name name;
+
+    /**
+     * Constuctor for MarkCommand
```
Comment (Incogdino) [inline src/main/java/seedu/address/logic/commands/UnmarkCommand.java:37]: I think there needs to be an empty line between the description and the parameters. Refer to the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#comments) for more info.

File: src/main/java/seedu/address/logic/parser/MarkCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import seedu.address.model.person.Name;
+
+/**
+ * Parses input arguments and creates a new MarkCommand object
+ */
+public class MarkCommandParser implements Parser<MarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the seedu.address.logic.commands.MarkCommand
+     * and returns a seedu.address.logic.commands.MarkCommand object for execution.
```
Comment (Incogdino) [inline src/main/java/seedu/address/logic/parser/MarkCommandParser.java:20]: I think there needs to be an empty line between the description and the parameters. Refer to the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#comments) for more info.

File: src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import seedu.address.model.person.Name;
+
+/**
+ * Parses input arguments and creates a new MarkCommand object
+ */
+public class UnmarkCommandParser implements Parser<UnmarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the seedu.address.logic.commands.MarkCommand
+     * and returns a seedu.address.logic.commands.MarkCommand object for execution.
```
Comment (Incogdino) [inline src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java:20]: Same as the previous comments, might be better to leave a newline here.

File: src/main/java/seedu/address/model/Model.java [hunk 4]
```diff
@@ -83,6 +84,13 @@ public interface Model {
      */
     void setPerson(Person target, Person editedPerson);
 
+    /**
+     * Returns the person in the address book with the given name.
```
Comment (Incogdino) [inline src/main/java/seedu/address/model/Model.java:88]: Same as the other comments, might be better to leave a newline here.

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -74,6 +77,40 @@ public Person(
         this.address = address;
         this.tags.addAll(tags);
         this.assignment = assignment;
+        this.weeksPresent = new HashSet<>();
+    }
+
+    /**
+     * Constructs a new Person object
```
Comment (Incogdino) [inline src/main/java/seedu/address/model/person/Person.java:84]: Same as the other comments, might be better to leave a newline here.

File: src/main/java/seedu/address/model/person/Person.java [hunk 6]
```diff
@@ -74,6 +77,40 @@ public Person(
...
+     * @param name the name of the person
+     * @param phone the phone number of the person
+     * @param address the address of the person
```
Comment (Incogdino) [inline src/main/java/seedu/address/model/person/Person.java:87]: Can add full stops at the end of each parameter sentence.

File: src/main/java/seedu/address/model/person/Person.java [hunk 7]
```diff
@@ -116,6 +153,26 @@ public Github getGithub() {
         return github;
     }
 
+    public void markWeekPresent(int week) {
+        this.weeksPresent.add(week);
+    }
+
+    public void unmarkWeekPresent(int week) {
```
Comment (Incogdino) [inline src/main/java/seedu/address/model/person/Person.java:160]: This block of code seemed to be unused, could perhaps remove it if it is not needed.

File: src/test/java/seedu/address/logic/commands/GitHubCommandTest.java [hunk 8]
```diff
@@ -90,7 +91,7 @@ public void execute_nameNotInModel_throwsCommandException() throws Exception {
     /**
      * A default model stub that have all the methods failing.
      */
-    private class ModelStub implements Model {
+    private abstract class ModelStub implements Model {
         @Override
         public ReadOnlyUserPrefs getUserPrefs() {
```
Comment (Incogdino) [inline src/test/java/seedu/address/logic/commands/GitHubCommandTest.java:96]: Good use of abstract class here.

**Review by Incogdino — COMMENTED** [review]
LGTM, just some comments on javadoc.

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_WEEK + "1";
+
+    public final int week;
+    private final Name name;
+
+    /**
+     * Constuctor for UnmarkCommand
+     *
+     * @param name the name of the student
```
Comment (tohjh) [inline src/main/java/seedu/address/logic/commands/UnmarkCommand.java:40]: Can consider having more details such as:
@param name the name of the student to be unmarked
@param week the week number for which the student is being unmarked as present

File: src/main/java/seedu/address/logic/commands/MarkCommand.java [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_WEEK + "1";
+
+    public final int week;
+    private final Name name;
+
+    /**
+     * Constuctor for MarkCommand
+     *
+     * @param name the name of the student
```
Comment (tohjh) [inline src/main/java/seedu/address/logic/commands/MarkCommand.java:39]: Same as unmark, perhaps could have more details for the parameters

File: src/main/java/seedu/address/model/Model.java [hunk 11]
```diff
@@ -84,6 +85,14 @@ public interface Model {
...
+     *
+     * @param name The Name to search for
```
Comment (tohjh) [inline src/main/java/seedu/address/model/Model.java:91]: The capitalization seem a bit weird

File: src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Parses input arguments and creates a new UnmarkCommand object
+ */
+public class UnmarkCommandParser implements Parser<UnmarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the seedu.address.logic.commands.UnmarkCommand
+     * and returns a seedu.address.logic.commands.UnmarkCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (tohjh) [inline src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java:22]: Might want to add full stop for consistency as the java docs above does

File: src/main/java/seedu/address/logic/commands/MarkCommand.java [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+     * @param week the week number
```
Comment (tohjh) [inline src/main/java/seedu/address/logic/commands/MarkCommand.java:40]: Some param have full stop some don't have. Might want to make it consistent

**Review by DesSnowy — APPROVED** [review]
LGTM!

**Pull request merged**